Tests/NIOTests/ChannelTests.swift

```diff
@@ -2200,4 +2200,98 @@ public class ChannelTests: XCTestCase {
         XCTAssertNoThrow(try allDone.futureResult.wait())
         XCTAssertNoThrow(try sc.syncCloseAcceptingAlreadyClosed())
     }
+
+    func testConnectWithECONNRESETGetsTheRightError() throws {
+        let group = MultiThreadedEventLoopGroup(numThreads: 1)
+        defer {
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+        let serverSockSyncQ = DispatchQueue(label: "serverSock syncQ")
+        let serverSock = try serverSockSyncQ.sync { () -> Socket in
+            let sock = try Socket(protocolFamily: PF_INET, type: Posix.SOCK_STREAM)
+            XCTAssertNoThrow(try sock.bind(to: SocketAddress(ipAddress: "127.0.0.1", port: 0)))
+            XCTAssertNoThrow(try sock.withUnsafeFileDescriptor { fd in
+                try Posix.listen(descriptor: fd, backlog: 1)
+            })
+            return sock
+        }
+        let serverSockAddress = serverSockSyncQ.sync { try! serverSock.localAddress() }
+        defer {
+            XCTAssertNoThrow(try serverSockSyncQ.sync { try serverSock.close() })
+        }
+        let g = DispatchGroup()
+        let q = DispatchQueue(label: "q")
+        do {
+            // spawn 64 connects, likely some will already get ECONNREFUSED as the backlog isn't large enough
+            let fs = (0..<64).map { _ in ClientBootstrap(group: group).connect(to: serverSockAddress) }
+            q.async(group: g) { () -> Void in
+                while true {
+                    do {
+                        // try to accept one client
+                        let accepted = try serverSockSyncQ.sync { () -> CInt? in
+                            try serverSock.withUnsafeFileDescriptor { fd in
+                                var pfd = pollfd(fd: fd, events: Int16(POLLIN | POLLERR | POLLHUP), revents: 0)
+                                // wait a maximum of 100ms to accept a new client, just to be guaranteed we don't get stuck here.
+                                if poll(&pfd, 1, 100) == 1 {
+                                    return accept(fd, nil, nil)
```

Comment:
Any reason not to just block in `accept` here?

Reply:
yeah, that would hang forever so we need the 100ms timeout here

Reply:
In what circumstance will it hang forever?

Reply:
@Lukasa well, it calls `accept` on a socket. That would only ever return if there was someone connecting. But if there are no further connections (remember the 64 client connections each might either fail or (at least partially) succeed)